examples/federated/bootstrap.js

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+// This is all the data needed to load and activate plugins. This should be
+// gathered by the server and put onto the initial page template.
+const PLUGIN_DATA = JSON.parse(document.getElementById('jupyterlab-plugin-data').textContent);
+
+import { PageConfig } from '@jupyterlab/coreutils';
+// eslint-disable-next-line
+__webpack_public_path__ = PageConfig.getOption('fullStaticUrl') + '/';
+
+// This must be after the public path is set.
+// This cannot be extracted because the public path is dynamic.
+require('./build/app/imports.css');
+
+function loadScript(url) {
+  return new Promise((resolve, reject) => {
+    const newScript = document.createElement('script');
+    newScript.onerror = reject;
+    newScript.onload = resolve;
+    newScript.async = true;
+    document.head.appendChild(newScript);
+    newScript.src = url;
+  });
+}
+
+async function loadComponent(url, scope, module) {
+  await loadScript(url);
+
+  // From MIT-licensed https://github.com/module-federation/module-federation-examples/blob/af043acd6be1718ee195b2511adf6011fba4233c/advanced-api/dynamic-remotes/app1/src/App.js#L6-L12
+  await __webpack_init_sharing__('default');
+  const container = window.MYNAMESPACE[scope];
+  // Initialize the container, it may provide shared modules and may need ours
+  await container.init(__webpack_share_scopes__.default);
```

Comment:
If we have a number of dynamically loaded extensions, we may want to do this container initialization first (I think this is what Tobias was getting at?), and *then* do the the `.get()` call below for each module. I think (hope?) this will allow the deduplicating logic to run as intended.